### PR TITLE
Implement GraphQL functionality

### PR DESCRIPTION
**Pull request by kq-li**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,16 +1,40 @@
+from datetime import time
+
```
Comment (young-k) [inline app.py]: delete this

File: schema.py [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+from graphene import ObjectType, String, ID, List
+from graphene.types.datetime import DateTime
+
+class GymType(ObjectType):
+  id = ID()
+  name = String()
+  hours = String()
 
 class Query(ObjectType):
   hello = String(name=String(required=True))
```
Comment (young-k) [inline schema.py]: why is this here, is it just leftover from the original stuff?

File: schema.py [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
     return 'Hello %s!' % name
+
+  @staticmethod
+  def check_gym_open(now, gym):
+    for (weekday, open_time, close_time) in gym['times']:
+      if weekday == now.weekday() and now.time() > open_time and now.time() < close_time:
+        return True
+    return False
+
+  def resolve_gyms(self, info, now=None):
```
[the remark this answers is not in the document]
Comment (kq-li) [reply schema.py]: this is to support both querying for time and for all gyms through graphql

**Comment by kq-li** [comment]
note that this is very much a WIP and i'm planning on rebasing, added reviewers but for now feel free to ignore -- will ping closer to completion

File: app.py (hunk 1 above)
Comment (kq-li) [reply app.py]: used in line 17

File: schema.py (hunk 2 above)
Comment (kq-li) [reply schema.py]: forgot to take it out, was a test

File: app.py (hunk 1 above)
Comment (young-k) [reply app.py]: i meant the newline lol

**Comment by young-k** [comment]
Once you have this working, can you add some screenshots of queries that you can run?

**Comment by kq-li** [comment]
ready for review, here's some queries

query1
query2
query3
query4
query5
query6
query7
query8
query9

File: schema.py [hunk 4]
```diff
@@ -1,7 +1,127 @@
...
+  gym = Field(GymType)
+  details_id = ID()
+  details = Field(ClassDetailType)
+  start_time = DateTime()
+  end_time = DateTime()
+  instructor = String()
+  is_cancelled = Boolean()
+
+  def resolve_gym(self, info):
+    return gyms.get(self.gym_id)
```
Comment (DanielNosrati) [inline schema.py]: Is info supposed to be used. Also i feel slightly akward about modifying global state in the way that this is being done.

Comment (kq-li) [reply schema.py]: the resolvers are part of the library and get passed info by default
i'm using global state here just as an in-memory store to bypass the need for a database -- the design is a little iffy but i'm not sure of a better way to do it, open to suggestions
